File: crates/infergen-core/templates/django/snippet.py
# After `infergen generate`, import your typed SDK:
from infergen_generated import track

# In a view:
def my_view(request):
    track.view_requested(view_name="my_view", method=request.method)
    # ...

# In a login view:
def login_view(request):
    success = authenticate(request)
    track.user_login_attempted(success=success)

File: crates/infergen-core/templates/express/snippet.ts
// After `infergen generate`, import your typed SDK:
import { track } from "./infergen.generated";

app.use((req, res, next) => {
  track.apiRequestReceived({ method: req.method, path: req.path });
  next();
});

app.post("/login", (req, res) => {
  // after auth attempt:
  track.userLoginAttempted({ success: true });
  res.json({ ok: true });
});
